src/Collection/SortableIterator.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Copyright (c)2015-2015 heiglandreas
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIBILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @category
+ * @author    Andreas Heigl<andreas@heigl.org>
+ * @copyright ©2015-2015 Andreas Heigl
+ * @license   http://www.opesource.org/licenses/mit-license.php MIT-License
+ * @version   0.0
+ * @since     22.07.15
+ * @link      https://github.com/heiglandreas/
+ */
+
+namespace Zend\Ldap\Collection;
+
+use Zend\Ldap\Ldap;
+
+class SortableIterator extends DefaultIterator
+{
+    /**
+     * @var array $entries
+     */
+    protected $entries = array();
+
+    public function __construct(Ldap $ldap, $resultId)
+    {
+        parent::__construct($ldap, $resultId);
+
+        $identifier = ldap_first_entry(
+            $ldap->getResource(),
+            $resultId
+        );
+
+        $identifier;
+
+        while (false !== $identifier) {
+
+            $this->entries[] = [
+                'resource' => $identifier,
+                'sortValue' => '',
+            ];
+
+            $identifier = ldap_next_entry(
+                $ldap->getResource(),
+                $identifier
+            );
+        }
+
+
+    }
+
+    /**
+     * Sort the iterator by the given attribute
+     *
+     * @param string $sortAttribute
+     *
+     * @return self
+     */
+    public function sort($sortAttribute)
+    {
+        foreach ($this->entries as $key => $entry) {
+
+            $attributes = ldap_get_attributes(
+                $this->ldap->getResource(),
+                $entry['resource']
+            );
+
+            if (isset($attributes[$sortAttribute][0])) {
+                $this->entries[$key]['sortValue'] = $attributes[$sortAttribute][0];
+            }
+        }
+
+        return usort($this->entries, function($a,$b){
+            return strnatcasecmp($a['sortValue'], $b['sortValue']);
+        });
```

Comment:
I like the idea of inject a custom sort logic.

But if I insert my own logic then `$sortAttribute` could be optional? If yes then I think should be two different methods.